src/main/java/com/google/sps/servlets/PostAnswerServlet.java

```diff
@@ -0,0 +1,121 @@
+      logger.log(Level.SEVERE, exception.getMessage(), exception);
+    }
+    response.sendRedirect("/question.html?id=" + question_id);
+  }
+
+  /** 
+   * Inserts an answer into the database.
+   */
+  private void insertNewAnswer(Connection connection, int question_id, String body, int author_id) {
+    try {
+      String insertAnswerQuery = "INSERT INTO Answer(question_id, body, author_id, date_time) "
+          + "VALUES (?,?,?,NOW())";
+      PreparedStatement answerStatement = connection.prepareStatement(insertAnswerQuery);
+      answerStatement.setInt(SqlConstants.ANSWER_INSERT_QUESTIONID_COLUMN, question_id);
+      answerStatement.setString(SqlConstants.ANSWER_INSERT_BODY_COLUMN, body);
+      answerStatement.setInt(SqlConstants.ANSWER_INSERT_ASKERID_COLUMN, author_id);
+      answerStatement.executeUpdate();
+    } catch (SQLException exception) {
+      // If the connection or the query don't go through, we get the log of what happened.
+      Logger logger = Logger.getLogger(PostAnswerServlet.class.getName());
+      logger.log(Level.SEVERE, exception.getMessage(), exception);
+    }
+  }
+
+  /** 
+   * Gets the ID from the last answer posted. Returns -1 on query failure.
+   */
+  private int getLatestAnswerId(Connection connection) {
+    int id = -1;
+    try {
+      String maxIdQuery = "SELECT MAX(id) FROM Answer;";
+      PreparedStatement maxIdStatement = connection.prepareStatement(maxIdQuery);
+      ResultSet queryResult = maxIdStatement.executeQuery();
+      queryResult.next();
+      id = queryResult.getInt(SqlConstants.ANSWER_FETCH_MAXID_COLUMN);
+    } catch (SQLException exception) {
+      // If the connection or the query don't go through, we get the log of what happened.
+      Logger logger = Logger.getLogger(PostAnswerServlet.class.getName());
+      logger.log(Level.SEVERE, exception.getMessage(), exception);
+    }
+
+    return id;
+  }
+
+  /** 
+   * Makes the author of the recently added answer a follower of said answer.
+   * TODO(shaargtz): Move function to Utiliy class to be reused.
+   */
+  private void insertNewFollower(Connection connection, int author_id) {
+    try {
+      int latestAnswerId = getLatestAnswerId(connection);
+      String insertFollowerQuery = "INSERT INTO AnswerFollower(answer_id, follower_id) "
+          + "VALUES (?,?)";
+      PreparedStatement followerStatement = connection.prepareStatement(insertFollowerQuery);
+      followerStatement.setInt(SqlConstants.FOLLOWER_INSERT_ANSWERID_COLUMN, latestAnswerId);
+      followerStatement.setInt(SqlConstants.FOLLOWER_INSERT_AUTHORID_COLUMN, author_id);
+      followerStatement.executeUpdate();
+    } catch (SQLException exception) {
+      // If the connection or the query don't go through, we get the log of what happened.
+      Logger logger = Logger.getLogger(PostAnswerServlet.class.getName());
+      logger.log(Level.SEVERE, exception.getMessage(), exception);
+    }
+  }
+}
```

Comment:
Why have a separate PostAnswerServlet and FetchAnswersServlet? I think it would be better to have just the AnswerServlet, and have both the doGet() and doPost() functions in there.

Reply:
i think that is a pretty good idea, and it could also be done with the question servlets, I will leave a TODO since I have stuff dependent on this branch and I don't want some really bad merge conflicts ahahah

Reply:
Great, please do!